contrib/epee/include/span.h

```diff
@@ -81,6 +88,13 @@ namespace epee
     constexpr span(const span&) noexcept = default;
     span& operator=(const span&) noexcept = default;
 
+    void remove_prefix(std::size_t amount) noexcept
+    {
+        amount = std::min(len, amount);
+        ptr += amount;
+        len -= amount;
+    }
```

Comment:
Should this not error out if it cannot remove the amount required ?

Reply:
As an exception or bool? A majority of the time, clamping to the end without exception makes sense. There might be some infinite loop bug aborted by an exception here. Usually the bug is `remove_prefix(0)` though.

Reply:
Any method really. It just seems like a bug at first glance since it doesn't seem obvious unless you read the code that it can do less than asked. things like read that can do short reads return the number of bytes read, for instance.

Reply:
I like the number of bytes shifted approach actually, its fairly close to many other interfaces. Will make this change shortly.

Reply:
You might have forgot about this.

Reply:
I completed this at the time but forgot to push out the change. Crud. Just forced push the change.